### PR TITLE
feat: Add private-endpoint e2e

### PR DESCRIPTION
**Pull request by segraef**
## Description

Added e2e test with private-endpoint to reflect https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/deployment-script-vnet-private-endpoint.

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.resources.deployment-script](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.resources.deployment-script.yml)     |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/resources/deployment-script/version.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
-  "version": "0.2",
+  "version": "0.3",
```
Comment (AlexanderSehr) [inline avm/res/resources/deployment-script/version.json:3]: Hey @segraef, it's ultimately up to @sebassem, but I just got to ask: Why the version.json update? 😄 From the looks of it, it's just a new test that is added, right?

File: avm/res/resources/deployment-script/tests/e2e/private-endpoint/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+@description('Required. The name of the managed identity to create.')
+param managedIdentityName string
+
+@description('Required. The name of the Virtual Network to create.')
+param virtualNetworkName string
+
+@description('Required. The name of the Storage Account to create.')
+param storageAccountName string
+
+param privateEndpointName string
```
Comment (AlexanderSehr) [inline avm/res/resources/deployment-script/tests/e2e/private-endpoint/dependencies.bicep:13]: @description('Required. The name of the Private Endpoint to create.')

File: avm/res/resources/deployment-script/tests/e2e/private-endpoint/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+resource storageAccount 'Microsoft.Storage/storageAccounts@2023-04-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  kind: 'StorageV2'
+  properties: {
+    publicNetworkAccess: 'Disabled'
+    // allowSharedKeyAccess: true // Cannot be set to false when using a private network for the deployment script
```
Comment (AlexanderSehr) [inline avm/res/resources/deployment-script/tests/e2e/private-endpoint/dependencies.bicep:47]: Leftover?

File: avm/res/resources/deployment-script/tests/e2e/private-endpoint/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      ]
+    }
+    timeout: 'PT1H'
+    runOnce: true
+    scriptContent: 'echo \'AVM Deployment Script test!\''
+    storageAccountResourceId: nestedDependencies.outputs.storageAccountResourceId
+  }
+  dependsOn: [
+    nestedDependencies
+  ]
```
Comment (AlexanderSehr) [inline avm/res/resources/deployment-script/tests/e2e/private-endpoint/main.test.bicep:75]:
```suggestion
```

File: avm/res/resources/deployment-script/version.json (hunk 1 above)
Comment (segraef) [reply avm/res/resources/deployment-script/version.json:3]: This triggered me

image

so yes you're right we can keep `minor` and let `patch` increase automatically.

**Comment by segraef** [comment]
Re-triggered: [avm.res.resources.deployment-script](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.resources.deployment-script.yml)

File: avm/res/resources/deployment-script/main.json [hunk 5]
```diff
@@ -492,4 +492,4 @@
       "value": "[split(reference('deploymentScriptLogs').log, '\n')]"
     }
   }
-}
\ No newline at end of file
+}
```
Comment (AlexanderSehr) [inline avm/res/resources/deployment-script/main.json:495]: Not sure what's happening here, must be the line ending just as with the `version.json` file. 

Just a note on that for awareness: If this PR goes in as is, the publishing step will run a `git diff` & detect that there was a change in both of these files. This will cause it to decide to publish a new version.
If they would not be part of the PR, the diff would only include the test files & readme, which cause the publishing script to ignore them and not publish a new version.

Long story short - I'd recommend to update the ending to their original form in both files 😏

Comment (segraef) [reply avm/res/resources/deployment-script/main.json:495]: So what happens is `Set-AVM-Readme` does not create an empty line during the generation of the `main.json` to signalise EOF  but then `bicep build` creates an empty line which then overwrites the `main.json` during the `New-AzDeployment`.

image

Sometimes when you test locally it might fail or get stuck leaving the `main.json` with an EOL resulting in this when one forgets to run `Set-AVMReadme` again to generate a `main.json` without an empty line.

**Comment by segraef** [comment]
@sebassem let me know if you're fine with this additional e2e test.

File: avm/res/resources/deployment-script/main.json (hunk 5 above)
Comment (AlexanderSehr) [reply avm/res/resources/deployment-script/main.json]: Hey @segraef, that's fair - and hopefully something we can somehow fix in the utility.
But for the time being, can you undo the change in the `version.json` file? If you don't the CI **will** publish a new version of the module - and given that the purpose of the PR was to add a test, I'm not sure that's your intention 😄

File: avm/res/resources/deployment-script/version.json [hunk 6]
```diff
@@ -4,4 +4,4 @@
   "pathFilters": [
     "./main.json"
   ]
-}
\ No newline at end of file
+}
```
Comment (AlexanderSehr) [inline avm/res/resources/deployment-script/version.json:7]: Still needs to be undone

File: avm/res/resources/deployment-script/main.json (hunk 5 above)
Comment (AlexanderSehr) [reply avm/res/resources/deployment-script/main.json]: Still needs to be undone 😉

**Comment by sebassem** [comment]
LGTM

**Review by sebassem — DISMISSED** [review]
LGTM

**Comment by segraef** [comment]
Retriggered e2e: [avm.res.resources.deployment-script](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.resources.deployment-script.yml)

**Review by AlexanderSehr — APPROVED** [review]
